### PR TITLE
Allow shoot worker pull images from current project's GCR

### LINKED ISSUE #7: Enable shoot worker pull image from current project's GCR
GKE default service account is allowed to pull images from the current project's GCR which is very convenient. And users don't have to configure image pull secret statically which also enhances security. For Gardener managed shoot worker nodes, we should also provide this functionality.

### PR DESCRIPTION
**Pull request by tennix**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
This PR Fixes #7 

**Special notes for your reviewer**:

This PR binds the shoot worker's service account to GCR bucket object viewer permission which enables pull permission according to https://cloud.google.com/container-registry/docs/access-control#grant-bucket

> ROLE is the Cloud Storage role you want to grant. 
>  * objectViewer to pull images
>  * objectAdmin to push and pull images

**Release note**:

```improvement user
Allow shoot worker pull image from current project's GCR
```

### CODE REVIEW

**Comment by tennix** [comment]
I've tested this by setting gcr registry private, and the deploying a pod using the project's gcr image. The pod can be created successfully.

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.7.0-dev.6
+v1.7.0-dev.8
```
Comment (tennix) [inline VERSION:1]: The previous release tagged but did not change this version, so bump directly to `v1.7.0-dev.8`

File: charts/internal/gcp-infra/templates/main.tf [hunk 2]
```diff
@@ -14,6 +14,18 @@ resource "google_service_account" "serviceaccount" {
   display_name = "{{ required "clusterName is required" .Values.clusterName }}"
 }
 
+//=====================================================================
+//= Allow node pull images from current project's GCR
+//=====================================================================
+
+resource "google_storage_bucket_iam_binding" "binding" {
```
Comment (aylei) [inline charts/internal/gcp-infra/templates/main.tf:21]: should be `iam_member`, otherwise different shoots override each other

**Review by aylei — APPROVED** [review]
lgtm

**Comment by tennix** [comment]
@qiffang PTAL

**Review by qiffang — APPROVED** [review]
lgtm

**Pull request merged**